independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/strategy/ReplaceCodestartFileStrategyHandler.java

```diff
@@ -4,16 +4,23 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
 import java.util.List;
 import java.util.Map;
 
-final class ReplaceCodestartFileStrategyHandler implements CodestartFileStrategyHandler {
+final class ReplaceCodestartFileStrategyHandler implements DefaultCodestartFileStrategyHandler {
 
     @Override
     public String name() {
         return "replace";
     }
 
+    @Override
+    public void copyStaticFile(Path sourcePath, Path targetPath) throws IOException {
+        Files.createDirectories(targetPath.getParent());
```

Comment:
This throws `FileAlreadyExistsException` if `targetPath.getParent()` already exists

Reply:
@gastaldi yeah, only if the file exists **AND is not a directory**, which makes sense here:
"Creates a directory by creating all nonexistent parent directories first. Unlike the createDirectory method, an exception is **NOT** thrown if the directory could not be created because it already exists."